src/Microsoft.Extensions.Localization/Internal/AssemblyResourceStringProvider.cs

```diff
@@ -96,11 +95,14 @@ public IList<string> GetAllResourceStrings(CultureInfo culture, bool throwOnMiss
 
         protected virtual AssemblyWrapper GetAssembly(CultureInfo culture)
         {
-            var assemblyString = ApplyCultureToAssembly(culture);
             Assembly assembly;
             try
             {
-                assembly = Assembly.Load(new AssemblyName(assemblyString));
+                var assemblyName = new AssemblyName(_assembly.FullName)
```

Comment:
Move this out of the try-catch, only the code which will throw should be in there.